elastic/apm-server
# Set es-client config option enable_compatibility_mode for compatible ES responses

## Motivation/summary

https://github.com/elastic/go-elasticsearch/pull/355 introduced support for setting
a config option for the 7.x client to request 7.x compatible responses from Elasticsearch.
Reuse the supported config options for libbeat (setting the actual headers) to set this
es-client configuration option. The http headers need to be removed from the client as
the es-client takes care of setting them.


## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these change
* Configure es output headers in apm-server.yml 
```
output.elasticsearch:
    headers:
      Content-Type: "application/vnd.elasticsearch+json;compatible-with=7"
      Accept: "application/vnd.elasticsearch+json;compatible-with=7"
```
* Run apm-server `7.16` (not from this PR) and observe ingestion errors
* Run apm-server from this PR and observe that data are properly ingested (you can also add debugging and explicitly check that the `accept` and `content-type` headers are properly set) 




## Related issues

related to #6426
